### PR TITLE
Fix  #1923: Tests for Update Submission Data

### LINKED ISSUE #1923: Add test cases for Update submission data API
Since we have an API for updating submission data after a remote evaluation is completed, so we would like to have it's unit cases.
Link to the API: https://github.com/Cloud-CV/EvalAI/blob/master/apps/jobs/views.py#L583

### PR DESCRIPTION
**Pull request by Ayukha**
closes  #1923

### CODE REVIEW

File: tests/unit/jobs/test_views.py [hunk 1]
```diff
@@ -835,9 +837,118 @@ def test_get_submission_by_pk_when_user_is_neither_challenge_host_nor_submission
         self.url = reverse_lazy('jobs:get_submission_by_pk',
                                 kwargs={'submission_id': self.submission.id})
 
-        expected = {'error': 'Sorry, you are not authorized to access this submission.'}
+        expected = {
+            'error': 'Sorry, you are not authorized to access this submission.'}
```
Comment (RishabhJain2018) [inline tests/unit/jobs/test_views.py]: @Ayukha Can you please revert the changes above this because they are not required for this PR.

**Comment by RishabhJain2018** [comment]
@Ayukha Can you please resolve the conflicts ?

**Comment by RishabhJain2018** [comment]
@Ayukha Any updates on this?

**Comment by Ayukha** [comment]
@RishabhJain2018  Please review.

**Comment by RishabhJain2018** [comment]
@Ayukha Can you please fix the travis build?

**Comment by Ayukha** [comment]
@RishabhJain2018  Please review

**Comment by deshraj** [comment]
@RishabhJain2018 can you please review this PR? It has been around for a while.

**Comment by deshraj** [comment]
@RishabhJain2018 can you please review this PR? It has been around for a while. 

@Ayukha can you please resolve the conflict for one last time? Sorry for asking you to resolve conflicts again.

File: tests/unit/jobs/test_views.py [hunk 2]
```diff
@@ -1453,3 +1454,254 @@ def test_get_private_leaderboard_when_user_is_participant(self):
...
+            ])
+        }
+        expected = {
+            'success': 'Submission result has been successfully updated'}
+        self.client.force_authenticate(user=self.challenge_host.user)
+        response = self.client.put(self.url, self.data)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_submission_for_invalid_data(self):
```
Comment (RishabhJain2018) [inline tests/unit/jobs/test_views.py]: `test_update_submission_for_invalid_data` -> `test_update_submission_for_invalid_data_in_result_key`

File: tests/unit/jobs/test_views.py [hunk 3]
```diff
@@ -1453,3 +1454,254 @@ def test_get_private_leaderboard_when_user_is_participant(self):
...
+        }
+
+        expected = {
+            'error': '`result` key contains invalid data. Please try again with correct format!'}
+        self.client.force_authenticate(user=self.challenge_host.user)
+        response = self.client.put(self.url, self.data)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_submission_when_challenge_phase_split_not_exists(self):
```
Comment (RishabhJain2018) [inline tests/unit/jobs/test_views.py]: `test_update_submission_when_challenge_phase_split_not_exists` -> `test_update_submission_when_challenge_phase_split_not_exist`.

Comment (Ayukha) [reply tests/unit/jobs/test_views.py]: @RishabhJain2018  Changes done,Please review.
